src/Amalgam/AssetManager.cpp

```diff
@@ -111,65 +117,162 @@ void AssetManager::AssetParameters::UpdateResources()
 	}
 }
 
-EvaluableNodeReference AssetManager::LoadResource(AssetParameters &asset_params, EvaluableNodeManager *enm,
+//returns a pair of success followed by version number if a version can be found in an Amalgam metadata file
+static std::pair<bool, std::string> FindVersionStringInAmlgMetadata(std::ifstream &file)
+{
+	//read 200 bytes and null terminate
+	std::array<char, 201> buffer;
+	file.read(buffer.data(), 200);
+	buffer[200] = '\0';
+	std::string str(buffer.data(), 200);
```

Comment:
Is this sound if the file is shorter than 200 bytes?  In here (and probably the preceding line) do you need to use the return value from `file.read()` as the length/offset?